pkgs/development/beam-modules/elvis-erlang/default.nix

```diff
@@ -0,0 +1,56 @@
+{ fetchFromGitHub, fetchgit, fetchHex, rebar3WithPlugins, rebar3-nix, rebar3Relx
+, buildRebar3, writeScript, common-updater-scripts, coreutils, git, gnused
+, gnutar, gzip, nix, nix-prefetch-git, stdenv, lib }:
+let
+  version = "1.0.1";
+  owner = "inaka";
+  repo = "elvis";
+in rebar3Relx {
+  releaseType = "escript";
+  # The package name "elvis" is already taken
+  name = "elvis-erlang";
+  inherit version;
+  src = fetchFromGitHub {
+    inherit owner repo;
+    sha256 = "139mgd4cwc0vazxfnssyym61jd9g45wn1nc53mjfzx5dkrrn4dc5";
+    rev = version;
+  };
+  beamDeps = builtins.attrValues (import ./rebar-deps.nix {
+    inherit fetchHex fetchgit fetchFromGitHub;
+    builder = buildRebar3;
+  });
+  passthru.updateScript = writeScript "update.sh" ''
+    #!${stdenv.shell}
+    set -ox errexit
+    PATH=${
+      lib.makeBinPath [
+        common-updater-scripts
+        coreutils
+        git
+        nix-prefetch-git
+        nix
+        gnutar
+        gzip
+        (rebar3WithPlugins { globalPlugins = [ rebar3-nix ]; })
+      ]
+    }
```

Comment:
I think this could be simplified by using nix-shell.

Reply:
I'm curious about this too!
Are we talking about something like this?
```
#!/usr/bin/env nix-shell
#!nix-shell -i bash -p curl gnused nix-prefetch jq
```
The only problem I see with this, is the rebar3WithPlugins declaration with an attribute.

also tiny thing I think common-updater-scripts already includes `coreutils git nix-prefetch-git nix`

Reply:
> I think this could be simplified by using nix-shell.

Also curious about this, do you have an example of how it would work? My main concern with the #! approach is that a different version of nixpkgs would be used. In particular, `rebar3-nix` needs to be the right version.

> also tiny thing I think common-updater-scripts already includes `coreutils git nix-prefetch-git nix`

It looks like it makes them available to the scripts themselves, but not to our script, I tried it with `nix-shell --pure -p common-updater-scripts`

Reply:
https://github.com/NixOS/nixpkgs/blob/3ce1470717b9b0f772f6ff4441e7d3fabfabb047/doc/stdenv/stdenv.chapter.md#L283-L285

> My main concern with the #! approach is that a different version of nixpkgs would be used. In particular, rebar3-nix needs to be the right version.

The bot always uses a recent repo version of everything. Same can happen when you create the PR with a older git version.

Reply:
Interesting that the common-updater-scripts doesn't make those available, good to know!

Reply:
@SuperSandro2000 I thought about this a bit more and I think the `writeScript` approach might be the best way to go. I think it's a bit easier to reason about where the packages are coming from, and has the additional advantage that any updates which remove any of the updateScript's dependencies will cause an eval error (since with the nix-shell approach they would just be included as strings, unless I misunderstood the idea of course)

Reply:
> has the additional advantage that any updates which remove any of the updateScript's dependencies will cause an eval error (since with the nix-shell approach they would just be included as strings, unless I misunderstood the idea of course)

Don't worry about that. Everything except rebar3WithPlugins is used by hundreds or thousands of packages and a removal of it will always leave an alias behind.

Also when not using nix-shell doing an update on a state with many rebuilds is impossible without rebuilding half the world.

Reply:
Thanks for clarifying. Have updated the PR!